src/SimplePie.php

```diff
@@ -1774,25 +1777,29 @@ protected function fetch_data(&$cache)
                             $headers['if-none-match'] = $this->data['headers']['etag'];
                         }
 
-                        $file = $this->registry->create(File::class, [$this->feed_url, $this->timeout/10, 5, $headers, $this->useragent, $this->force_fsockopen, $this->curl_options]);
-                        $this->status_code = $file->status_code;
-
-                        if ($file->success) {
-                            if ($file->status_code === 304) {
-                                // Set raw_data to false here too, to signify that the cache
-                                // is still valid.
-                                $this->raw_data = false;
-                                $cache->set_data($cacheKey, $this->data, $this->cache_duration);
-                                return true;
-                            }
-                        } else {
+                        try {
+                            $orig_timeout = $this->timeout;
+                            $this->timeout = 1;
+                            $file = $this->get_http_client()->request(Client::METHOD_GET, $this->feed_url, $headers);
+                        } catch (HttpException $th) {
                             $this->check_modified = false;
                             if ($this->force_cache_fallback) {
                                 $cache->set_data($cacheKey, $this->data, $this->cache_duration);
                                 return true;
                             }
 
                             unset($file);
+                        } finally {
+                            $this->timeout = $orig_timeout;
+                        }
+                        $this->status_code = $file->status_code;
```

Comment:
Won’t `$file` be `null` when exception occurs and `$this->force_cache_fallback` is `false`?

Reply:
Yes, you are right. I will create a test for that.

Reply:
It was not possible for me to create a test for this complicated place at the moment, but I made sure $this->status_code is always set.